protocol/messenger.go

```diff
@@ -295,7 +292,7 @@ func NewMessenger(
 	encryptionProtocol := encryption.New(
 		database,
 		installationID,
-		c.onNewInstallationsHandler,
+		onNewInstallationsHandler,
```

Comment:
This can be a method instead.

Reply:
correct me if I am wrong, but this is relaying on the fact that `messenger` is a pointer (it's basically a closure), as it has not been initialized yet, so having it a method would work, but then we would need to pass a instantiated instance of messenger, which relies on encryption object , which in turns needs to be passed to messenger to instantiate :) 

so if I am not wrong we would have to first initialize messenger with some fields, then initialize encryptor passing the messenger function, and then set encryptor on messenger, which is a bit complicated, what do you think?

Reply:
Sorry, I did not notice that `onNewInstallationsHandler` is passed to the `encryption.New`. I though it's the `NewMessenger` constructor. You are right, this is good and less complicated. 

The only alternative I can think of is extracting the installations to a separate manager and share it with `encryptionProtocol` and `Messenger` but it does not sound like a task for now :)